src/OrbitGl/TimeGraph.cpp

```diff
@@ -564,6 +569,54 @@ void TimeGraph::ProcessCGroupAndProcessMemoryTrackingTimer(const TimerInfo& time
   }
 }
 
+void TimeGraph::ProcessPagefaultTrackingTimer(const orbit_client_protos::TimerInfo& timer_info) {
+  uint64_t cgroup_name_hash = timer_info.registers(
+      static_cast<size_t>(CaptureEventProcessor::PagefaultEncodingIndex::kCGroupNameHash));
+  std::string cgroup_name = app_->GetStringManager()->Get(cgroup_name_hash).value_or("");
+  int64_t system_pagefault = orbit_api::Decode<int64_t>(timer_info.registers(
+      static_cast<size_t>(CaptureEventProcessor::PagefaultEncodingIndex::kSystemPagefault)));
+  int64_t system_major_pagefault = orbit_api::Decode<int64_t>(timer_info.registers(
+      static_cast<size_t>(CaptureEventProcessor::PagefaultEncodingIndex::kSystemMajorPagefault)));
+  int64_t cgroup_pagefault = orbit_api::Decode<int64_t>(timer_info.registers(
+      static_cast<size_t>(CaptureEventProcessor::PagefaultEncodingIndex::kCGroupPagefault)));
+  int64_t cgroup_major_pagefault = orbit_api::Decode<int64_t>(timer_info.registers(
+      static_cast<size_t>(CaptureEventProcessor::PagefaultEncodingIndex::kCGroupMajorPagefault)));
+  int64_t process_minor_pagefault = orbit_api::Decode<int64_t>(timer_info.registers(
+      static_cast<size_t>(CaptureEventProcessor::PagefaultEncodingIndex::kProcessMinorPagefault)));
+  int64_t process_major_pagefault = orbit_api::Decode<int64_t>(timer_info.registers(
+      static_cast<size_t>(CaptureEventProcessor::PagefaultEncodingIndex::kProcessMajorPagefault)));
+
+  orbit_gl::PagefaultTrack* track = track_manager_->GetPagefaultTrack();
+  if (track == nullptr) {
+    const std::array<std::string, 3> kSeriesNames = {
+        absl::StrFormat("Process [%s]", capture_data_->process_name()),
+        absl::StrFormat("CGroup [%s]", cgroup_name), "System"};
+    track = track_manager_->CreateAndGetPagefaultTrack(kSeriesNames);
+  }
+
+  if (system_major_pagefault != kMissingInfo && cgroup_major_pagefault != kMissingInfo &&
+      process_major_pagefault != kMissingInfo) {
+    auto major_pagefault_track = track->GetMajorPagefaultTrack();
+    major_pagefault_track->AddValuesAndUpdateAnnotations(
+        timer_info.start(),
+        {static_cast<double>(process_major_pagefault), static_cast<double>(cgroup_major_pagefault),
+         static_cast<double>(system_major_pagefault)});
+    major_pagefault_track->SetIndexOfSeriesToHighlight(0);
```

Comment:
Why this in every process timer? I mean, could be in the creation.

Reply:
Moved to the `CreateAndGet` method. Thanks!